### PR TITLE
[QA-1861] Integration test failure screenshot

### PR DESCRIPTION
**Pull request by aweng98**
https://broadworkbench.atlassian.net/browse/DC-306

Save page screenshot every time test fails.

### CODE REVIEW

File: integration-tests/utils/integration-utils.js [hunk 1]
```diff
@@ -259,12 +261,17 @@ const openError = async page => {
 }
 
 const maybeSaveScreenshot = async (page, testName) => {
-  if (!screenshotDirPath) { return }
+  const dir = screenshotDirPath ?
+    screenshotDirPath :
+    process.env.SCREENSHOT_DIR || process.env.LOG_DIR || resolve(__dirname, '../test-results/screenshots')
```
Comment (aweng98) [inline integration-tests/utils/integration-utils.js:266]: Provide a default dir for screenshots. I tend to forget manually setting `SCREENSHOT_DIR` every day.

File: integration-tests/utils/integration-utils.js [hunk 2]
```diff
@@ -288,7 +295,6 @@ const withScreenshot = _.curry((testName, fn) => async options => {
   try {
     return await fn(options)
   } catch (e) {
-    await maybeSaveScreenshot(options.page, testName)
```
Comment (aweng98) [inline integration-tests/utils/integration-utils.js]: Don't need `withScreenshot()` anymore but I'm going to leave it unchanged for now. I found one scenario where this doesn't work, it fails to take screenshot if test had exceeded Jest test timeout.
Screen Shot 2022-05-10 at 10 57 25 AM

File: integration-tests/jest-circus-environment.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+class JestCircusEnvironment extends PuppeteerEnvironment {
+  constructor(config, context) {
+    super(config, context)
+    this.testName = parse(context.testPath).name
+  }
+
+  async handleTestEvent(event, state) {
+    const { name } = event
+    if (['hook_failure', 'test_fn_failure'].includes(name)) {
```
Comment (petesantos) [inline integration-tests/jest-circus-environment.js:16]: Is there any doc for this API? I looked around but couldn't find any.

**Review by petesantos — APPROVED** [review]

File: yarn.lock [hunk 4]
```diff
@@ -8706,7 +8706,7 @@ __metadata:
   languageName: node
   linkType: hard
 
-"jest-environment-puppeteer@npm:^6.0.3":
+"jest-environment-puppeteer@npm:^6.0.2, jest-environment-puppeteer@npm:^6.0.3":
```
Comment (cahrens) [inline yarn.lock]: What is requiring 6.0.2?

File: integration-tests/jest-circus-environment.js (hunk 3 above)
Comment (aweng98) [reply integration-tests/jest-circus-environment.js:16]: It's `jest-circus` runner, the default test runner for `jest` . https://github.com/facebook/jest/blob/main/packages/jest-circus/README.md#overview
I'll add this comment.

**Review by cahrens — APPROVED** [review]

File: yarn.lock (hunk 4 above)
Comment (aweng98) [reply yarn.lock]: I don't think anything is requiring ver 6.0.2. I think I messed it up during `yarn add`. I upgraded version in package.json just now. Ver 6.0.2 is gone.

**Pull request merged**